ios/engine/KMEI/KeymanEngine/Classes/Migrations.swift

```diff
@@ -116,6 +116,11 @@ public enum Migrations {
 
     // Detect possible version matches.
     let userResources = Storage.active.userDefaults.userResources ?? []
+
+    // If there are no pre-existing resources and we need to detect a version, this is a fresh install.
+    if userResources.count == 0 {
+      return [Version.freshInstall]
```

Comment:
Why not use Version.fallback? Then you don't need to add another version string.

Reply:
I thought about it, but I like how this is much more explicit when debugging.  There's also the unlikely-but-possible chance that someone (somehow) actually still has a device with an ancient version of the app that could theoretically match the following line:

```
let legacy_resources = VersionResourceSet(version: Version.fallback, resources: [european])
```

It costs us pretty little to keep things clean by not assuming this case (an ancient version being updated) impossible.